openstack/antiddos/v1/antiddos/list_ddos_status.go

```diff
@@ -0,0 +1,99 @@
+package antiddos
+
+import (
+	"reflect"
+
+	"github.com/opentelekomcloud/gophertelekomcloud"
+	"github.com/opentelekomcloud/gophertelekomcloud/internal/extract"
+)
+
+type ListDDosStatusOpts struct {
+	// ID of an EIP
+	FloatingIpId string
+	// If this parameter is not used,
+	// the defense statuses of all ECSs are displayed in the Neutron-queried order by default.
+	Status string `q:"status"`
+	// Limit of number of returned results
+	Limit int `q:"limit"`
+	// Offset
+	Offset int `q:"offset"`
+	// IP address. Both IPv4 and IPv6 addresses are supported.
+	// For example, if you enter ?ip=192.168,
+	// the defense status of EIPs corresponding to 192.168.111.1 and 10.192.168.8 is returned.
+	Ip string `q:"ip"`
+}
+
+type DDosStatus struct {
+	// Floating IP address
+	FloatingIpAddress string `json:"floating_ip_address"`
+	// ID of an EIP
+	FloatingIpId string `json:"floating_ip_id"`
+	// EIP type.
+	NetworkType string `json:"network_type"`
+	// Defense status
+	Status string `json:"status"`
+}
+
+func ListDDosStatus(client *golangsdk.ServiceClient, opts ListDDosStatusOpts) ([]DDosStatus, error) {
+	q, err := golangsdk.BuildQueryString(&opts)
+	if err != nil {
+		return nil, err
+	}
+
+	// GET /v1/{project_id}/antiddos
+	raw, err := client.Get(client.ServiceURL("antiddos")+q.String(), nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var res ListStatusResponse
+	err = extract.Into(raw.Body, &res)
+	if err != nil {
+		return nil, err
+	}
+
+	return FilterDdosStatus(res.DDosStatus, opts)
+}
+
+type ListStatusResponse struct {
+	// Total number of EIPs
+	Total int `json:"total"`
+	// List of defense statuses
+	DDosStatus []DDosStatus `json:"ddosStatus"`
+}
+
+func FilterDdosStatus(ddosStatus []DDosStatus, opts ListDDosStatusOpts) ([]DDosStatus, error) {
+	var refinedDdosStatus []DDosStatus
+	var matched bool
+	m := map[string]interface{}{}
+
+	if opts.FloatingIpId != "" {
+		m["FloatingIpId"] = opts.FloatingIpId
+	}
+
+	if len(m) > 0 && len(ddosStatus) > 0 {
+		for _, ddosStatus := range ddosStatus {
+			matched = true
+
+			for key, value := range m {
+				if sVal := getStructField(&ddosStatus, key); !(sVal == value) {
+					matched = false
+				}
+			}
+
+			if matched {
+				refinedDdosStatus = append(refinedDdosStatus, ddosStatus)
+			}
+		}
+	} else {
+		refinedDdosStatus = ddosStatus
+	}
+
+	return refinedDdosStatus, nil
+}
```

Comment:
it's not my code so I just copy it...

Reply:
Assume my comment optional then

Reply:
Do you have a better solution for implementation?

Reply:
You can check each field explicitly, e.g.
```go
for _, ddosStatus := range ddosStatus {
	if ddosStatus.FloatingIpId != opts.FloatingIpId {
		continue
	}
	...
}
```

There are just a few fields so it won't produce readability problem and will be obvious from reading and debugging perspective. 

Other solution is to write stand-alone universal implementation which will include some debugging features like difference listing. But this will still require both structures to have exactly same name for the same fields (or other mapping means, e.g. same JSON and query fields naming) - and this is far from obvious when maintaining this code.

Reply:
https://docs.otc.t-systems.com/api/antiddos/antiddos_02_0023.html

This function of querying by FloatingIpId does not come with the API, but is a function implemented by the SDK itself. Do we need to leave it? Or is it really useful?

@outcatcher @anton-sidelnikov

Reply:
That's the idea behind client-side filtering :) Checks like that should be done for each field checked on the client side only.